forte/data/readers/classification_reader.py

```diff
@@ -0,0 +1,239 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# mypy: ignore-errors
+import csv
+import importlib
+from typing import Iterator, Tuple, List
+from collections import OrderedDict
+from forte.common import Resources, ProcessorConfigError
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.base_reader import PackReader
+from ft.onto.base_ontology import *
+
+
+__all__ = ["ClassificationDatasetReader"]
+
+
+class ClassificationDatasetReader(PackReader):
+    r"""
+    A generic dataset reader class for classification dataset.
+    In general, it works with any classification tasks that
+    require concatenating a sequence of strings from dataset and predicting a class.
+
+
+    The user must specify the index2label mapping, data fields and subtext fields
+    and other parameters in configs based on their tasks in order to initialize
+    the dataset reader properly.
+    Index2label mapping is a dictionary with indices as keys and string labels as values,
+    it helps encoding and decoding string labels.
+    Data fields is a list of ontology paths representing data fields in the dataset.
+    They must have the same length and align with each other.
+    Subtext fields is also a list of ontology paths that represent data fields that
+     will be concatenated into input strings in the same order as the list.
```

Comment:
I think for first time users, it is very hard for them to understand what data fields and subtext fields are by reading the documentation alone, can you try to explain with simple terms, more details, and with immediate examples?